src/containers/main.tsx

```diff
@@ -4,40 +4,114 @@ import Quote from "../components/quote";
+            console.log(this.state);
+        });
 
-    "Proin sed nibh ac ipsum feugiat fermentum. Aenean a lacus vehicula, porttitor nisi non, placerat turpis. Pellentesque blandit maximus posuere. Sed porttitor ac turpis ac ullamcorper. Suspendisse a urna tortor. Donec efficitur odio et semper egestas. Nulla augue mi, imperdiet ut orci vitae, tincidunt commodo ipsum. Pellentesque ultrices ligula nisl, non hendrerit metus laoreet in. Praesent non tortor elit. Proin ac mi venenatis, rhoncus ex in, convallis sem. In interdum elit ut rutrum auctor. Aenean ac aliquam nisl, vitae iaculis elit. Maecenas porta fringilla tortor a venenatis. Nunc ac gravida enim.",
+    }
 
-    "Fusce posuere facilisis lorem sed interdum. Ut in elit libero. Nam arcu dui, iaculis at ultrices a, eleifend aliquet tortor. Aliquam auctor tortor vel dolor hendrerit ornare. Ut vel aliquam sem. Etiam sit amet lacus eget libero mattis laoreet non consequat nulla. Nullam at mi ac mauris vestibulum tincidunt. Aliquam euismod est leo, vitae aliquam justo tempor eget. In quis arcu risus. Ut condimentum, mi sit amet tempor imperdiet, eros ex tempor lorem, vel congue nunc lorem posuere ante. Praesent eget diam purus. Proin lectus leo, ultricies in luctus commodo, pulvinar eu nunc. Integer ultrices faucibus dolor, ut porttitor augue luctus nec. Praesent sed diam id ante aliquet vulputate. Aliquam eu congue lacus, vitae convallis neque. Nunc cursus, tortor quis mattis sollicitudin, quam libero interdum quam, eget finibus lorem orci ac sem."
-];
+    onSearch(query : string) {
+        this.state.fetching = true;
+        if (query.length === 0) {
+            this.setState({
+                searchQuotes: [],
+                searching: false,
+                fetching: false
+            });
+            console.log(this.state);
+        } else {
+            this.api_handle.qdbQuoteFind({
+                query: query
+            }).then((quotes) => {
+                this.setState({
+                    searchQuotes: quotes,
+                    offset: 0,
+                    fetching: false,
+                    searching: true
+                });
+            });
+        }
+    }
 
+    onSubmit(quote : ClassifiedQuote) {
+        let payload = quote.message ? quote.message : quote.messages.join("\n");
+        this.api_handle.qdbQuotePost({
+            body: {
+                body: payload
+            }
+        });
+    }
 
-class Main extends Component<{}, {}> {
     render() {
-        const quotes = lipsum.map((body) =>
-                                    <Quote
-                                        id={"0"}
-                                        author={"mhuan13@gmail.com"}
-                                        body={body}
-                                        addedAt={"2017-05-01T18:41:37+00:00"}
-                                    />);
+        const quoteSource = this.state.searching ? this.state.searchQuotes : this.state.quotes;
```

Comment:
Might be useful to unpack state at the beginning of render, i.e.
```javascript
let {searching, searchQuotes, quotes, quoteSource, fetching} = this.state;
```
just for the sake of readability with all this `this.state.*`